hashing/sha256/sha256.go

```diff
@@ -1,43 +1,47 @@
 package sha256
 
 import (
-	"crypto/sha256"
+	sha256Lib "crypto/sha256"
 
 	"github.com/ElrondNetwork/elrond-go/hashing"
 )
 
-var _ hashing.Hasher = (*Sha256)(nil)
+var _ hashing.Hasher = (*sha256)(nil)
 
-var sha256EmptyHash []byte
+// sha256 is a sha256 implementation of the hasher interface.
+type sha256 struct {
+	emptyHash []byte
+}
 
-// Sha256 is a sha256 implementation of the hasher interface.
-type Sha256 struct {
+// NewSha256 initializes the empty hash and returns a new instance of the sha256 hasher
+func NewSha256() *sha256 {
+	return &sha256{
+		emptyHash: computeEmptyHash(),
+	}
 }
 
 // Compute takes a string, and returns the sha256 hash of that string
-func (sha Sha256) Compute(s string) []byte {
-	if len(s) == 0 && len(sha256EmptyHash) != 0 {
-		return sha.EmptyHash()
+func (s *sha256) Compute(str string) []byte {
+	if len(str) == 0 {
+		return s.emptyHash
```

Comment:
can you return a copy of emptyHash?
otherwise if caller changes the array you would provide wrong data to all subsequent calls.

Reply:
done. good catch